src-rs/build.rs

```diff
@@ -76,8 +76,8 @@ pub fn link_swift_package(package_name: &str, package_root: &str) {
     let swift_target_info = get_swift_target_info();
 
     println!(
-        "cargo:rustc-link-search=native={}.build/{}/{}",
+        "cargo:rustc-link-searchx={}.build/{}/{}",
         package_root, swift_target_info.target.unversioned_triple, profile
     );
-    println!("cargo:rustc-link-lib=static={}", package_name);
+    println!("cargo:rustc-link-lib={}", package_name);
```

Comment:
Won't this make `rustc` search for dynamic libs, not just static ones? If that's the case I think it should still be `static`, otherwise accommodations would likely need to be made for including dynamically linked libraries in the final binary.

Reply:
That is true however when I run with lib=static I get errors from the compiler stating it is unable to find the library. Only resolution I can find is removing the static= which allows for it to compile and it runs fine that way.

Reply:
Right now I'm more confused by the requirement for this. Any idea why it's not using the lib when it's linked in `networking` and needs manual linking in `wg-ondemand`?
<img width="760" alt="image" src="https://user-images.githubusercontent.com/14191578/191532000-c78daa97-3637-466b-ab56-a5997ca41f64.png">

Reply:
Yeah for some reason the search flag is not added to my rustc command in the wg-ondemand crate. That's my patch job right now since the static library is not actually linked to the native library.

Reply:
Ok I think I got the magic combo:
<img width="726" alt="image" src="https://user-images.githubusercontent.com/14191578/191538816-e476ccdc-f8a7-492b-a02c-f83d8e098313.png">
`native` is the correct type for searching, whereas the type for the library apparently doesn't matter - though I would like to keep it as`static`. Doing this ensures that the swift library is statically linked to `networking` and no build script is required in `wg-ondemand`.

Reply:
<img width="652" alt="Screen Shot 2022-09-21 at 11 02 45" src="https://user-images.githubusercontent.com/21245275/191540136-de5ff0bc-3fdd-438e-af3d-28b87a9dbd7c.png">
😭

Reply:
Interestingly the -L flag is there. This is the rustc command that is run
`rustc --crate-name networking --edition=2021 networking/src/lib.rs --error-format=json --json=diagnostic-rendered-ansi,artifacts,future-incompat --crate-type lib --emit=dep-info,metadata,link -C embed-bitcode=no -C split-debuginfo=unpacked -C debuginfo=2 -C metadata=f708e3224b0bb3da -C extra-filename=-f708e3224b0bb3da --out-dir /Users/haydenstith/Documents/Repos/wg-ondemand/target/debug/deps -C incremental=/Users/haydenstith/Documents/Repos/wg-ondemand/target/debug/incremental -L dependency=/Users/haydenstith/Documents/Repos/wg-ondemand/target/debug/deps --extern swift_rs=/Users/haydenstith/Documents/Repos/wg-ondemand/target/debug/deps/libswift_rs-7749d74038bef324.rmeta -L native=/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/macosx -L native=/usr/lib/swift -L native=./platform/mac/.build/arm64-apple-macosx/debug -l static=platformMac`

Reply:
Ah hold on a sec, I forgot that I also modified the call to `link_swift_package` so that it didn't use relative paths:
<img width="662" alt="image" src="https://user-images.githubusercontent.com/14191578/191541473-86f43ae1-3091-4cc4-8588-3f31e0db29aa.png">

Reply:
Yes, works perfectly now. I'll go ahead and commit the changes.